lib/services/search.js

```diff
@@ -0,0 +1,395 @@
+'use strict';
+
+const es = require('elasticsearch'),
+  bluebird = require('bluebird'),
+  _ = require('lodash'),
+  util = require('util'),
+  log = require('./log').withStandardPrefix(__filename),
+  endpoint = process.env.ELASTIC_HOST || '',
+  serverConfig = {
+    host: endpoint,
+    maxSockets: 500,
+    apiVersion: '1.7',
+    // log: ['error', 'trace'],
+    defer: require('../utils/defer')
+  };
+
+/**
+ * Test whether or not the client is connected
+ * to Elastic
+ *
+ * @param  {object} currentClient
+ * @returns {Promise}
+ */
+function healthCheck(currentClient) {
+  return currentClient.ping({
+    requestTimeout: 1000
+  }).then(function () {
+    log('info', 'Elasticsearch cluster is up!');
+    return bluebird.resolve();
+  }).catch(function (error) {
+    log('info', 'Elasticsearch cluster is down!');
+    return bluebird.reject(error);
+  });
+}
+
+/**
+ * Create an Elasticsearch index
```

Comment:
i talked about the diff between aliases + indices elsewhere. mappings are for defining how our data is indexed.